lua/gatekeeper/policylib.lua

```diff
@@ -281,3 +281,37 @@ function decision_web(policy, tx_rate_kib_sec, cap_expire_sec,
 		policy, tx_rate_kib_sec, tx_rate_kib_sec * 0.05, -- 5%
 		cap_expire_sec, next_renewal_ms, renewal_step_ms, false)
 end
+
+local function lpm_add_tbl8s(ip_addr, prefix_len, prefixes)
+	local prefix
+	local masked_ip
+
+	if prefix_len <= 24 then
+		return prefixes
+	end
+
+	masked_ip = lpmlib.ip_mask_addr(ip_addr, 24)
+	prefix = masked_ip .. "/24"
+	prefixes[prefix] = true
+
+	return prefixes
+end
+
+-- Estimate the number of rules and number of tbl8s in DPDK LPM library.
+function lpm_para_estimate(ipv4_file)
+	local num_rules = 0
+	local num_tbl8s = 0
+	local prefixes = {}
+
+	for line in io.lines(ipv4_file) do
+		local ip_addr, prefix_len = lpmlib.str_to_prefix(line)
+		num_rules = num_rules + 1
+		prefixes = lpm_add_tbl8s(ip_addr, prefix_len, prefixes)
+	end
+
+	for i, v in ipairs(prefixes) do
+		num_tbl8s = num_tbl8s + 1
+	end
```

Comment:
If `lpm_add_tbl8s()` returns 0 when the new prefix is not added, and 1 when it is, we can avoid this loop.

`lpm_add_tbl8s()` is already modifying the table `prefixes`, so there's no need to return it.